src/marqo/tensor_search/tensor_search.py

```diff
@@ -1517,56 +1525,66 @@ def get_query_vectors_from_jobs(
     """
     result: Dict[Qidx, List[float]] = defaultdict(list)
     for qidx, ptrs in qidx_to_job.items():
-
-        # vectors = job_to_vectors[ptrs.job_hash][ptrs.start_idx: ptrs.end_idx]
-
-        # qidx_to_vectors[qidx].append(vectors)
         q = queries[qidx]
         index_info = get_index_info(config=config, index_name=q.index)
-
-        ordered_queries = list(q.q.items()) if isinstance(q.q, dict) else None
-        if ordered_queries:
-            # multiple queries. We have to weight and combine them:
-            vectorised_ordered_queries = [
-                (get_content_vector(
-                    possible_jobs=qidx_to_job[qidx],
-                    jobs=jobs,
-                    job_to_vectors=job_to_vectors,
-                    treat_urls_as_images=index_info.index_settings[NsField.index_defaults][NsField.treat_urls_and_pointers_as_images],
-                    content=content),
-                 weight,
-                 content
-                ) for content, weight in ordered_queries
-            ]
-            # TODO how doe we ensure order?
-            weighted_vectors = [np.asarray(vec) * weight for vec, weight, content in vectorised_ordered_queries]
-
-            context_tensors = q.get_context_tensor()
-            if context_tensors is not None:
-                weighted_vectors += [np.asarray(v.vector) * v.weight for v in context_tensors]
-
-            try:
-                merged_vector = np.mean(weighted_vectors, axis=0)
-            except ValueError as e:
-                raise errors.InvalidArgError(f"The provided vectors are not in the same dimension of the index."
-                                             f"This causes the error when we do `numpy.mean()` over all the vectors.\n"
-                                             f"The original error is `{e}`.\n"
-                                             f"Please check `https://docs.marqo.ai/0.0.16/API-Reference/search/#context`.")
-
-            if index_info.index_settings['index_defaults']['normalize_embeddings']:
-                norm = np.linalg.norm(merged_vector, axis=-1, keepdims=True)
-                if norm > 0:
-                    merged_vector /= np.linalg.norm(merged_vector, axis=-1, keepdims=True)
-            result[qidx] = list(merged_vector)
-        else:
-            # result[qidx] = vectors[0]
+        # Normal single text query
+        if isinstance(q.q, str):
             result[qidx] = get_content_vector(
                 possible_jobs=qidx_to_job.get(qidx, []),
                 jobs=jobs,
                 job_to_vectors= job_to_vectors,
                 treat_urls_as_images=index_info.index_settings[NsField.index_defaults][NsField.treat_urls_and_pointers_as_images],
                 content=q.q
             )
+            continue
+
```

Comment:
Do we need an elif here to check the type of q and an else to raise an error?

Reply:
Could you elaborate on this? Currently this method has handling for if `q` is a `str`, `None`, or `dict`.

Reply:
Merge the q type check, `None`, `str`, `dict`, into a combined if else branch to make the logic clearer?
if isinstance(q.q, str): ..., elif isinstance(q.q, None): ..., etc

Reply:
I can do a hybrid of that, sure! I won't make separate if branches for None and dict, as they share some of the same logic (using context vectors)

Reply:
Done! Could you check if this is clearer?